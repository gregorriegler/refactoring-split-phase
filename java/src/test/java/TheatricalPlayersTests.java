import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;

import java.util.List;

import static org.approvaltests.Approvals.verify;

@UseReporter(JunitReporter.class)
public class TheatricalPlayersTests {

    @Test
    public void exampleStatement() {
        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance(new Play("Hamlet", "tragedy"), 55),
                new Performance(new Play("As You Like It", "comedy"), 35),
                new Performance(new Play("Othello", "tragedy"), 40)));

        TheatricalPlayers theatricalPlayers = new TheatricalPlayers();
        var result = theatricalPlayers.print(invoice);

        verify(result);
    }
}
