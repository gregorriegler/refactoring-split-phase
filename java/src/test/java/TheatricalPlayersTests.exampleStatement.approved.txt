Statement for BigCo
Amount owed is $1,600.00
You earned 47 credits
